core/src/main/scala/spinal/core/sim/package.scala

```diff
@@ -690,6 +728,12 @@ package object sim {
     }
   }
 
+  implicit class SimEquivVecSeqPimper[T <: Data, E](s: Vec[T])(implicit ev: T => SimEquiv { type SimEquivT = E }) extends SimEquiv {
```

Comment:
Witchcraft ! XD

Reply:
Yeah, it was the cleanest way I could come up with. I also found the associated type (`SimEquivT`) worked much better than using a generic for some reason.

If it helps, a plain English explanation: The class is generic over some Scala type `E` and a SpinalHDL data type `T`. Then it requests that there should be an implicitly defined method which takes `T` and can map it to a type which implements `SimEquiv` with `E` as the associated type.